app/customer.py

```diff
@@ -0,0 +1,66 @@
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list[int, int],
+            money: int | float
+    ) -> None:
+        self.name = name
+        self.product_cart = product_cart
+        self.home_location = location
+        self.location = self.home_location
+        self.money = money
+        self.trip_costs = {}
+        self.chosen_shop = None
+        self.spent_money = 0
+
+    def amount_of_money_at_start(self) -> None:
+        print(f"{self.name} has {self.money} dollars")
+
+    def cost_of_all_trips(
+            self,
+            stores: list[dict],
+            fuel_consumption: float,
+            fuel_price: float
+    ) -> None:
+        for store in stores:
+            shop = Shop(
+                name=store["name"],
+                location=store["location"],
+                products=store["products"]
+            )
+            distance = (((shop.location[0] - self.location[0]) ** 2)
+                        + ((shop.location[1] - self.location[1]) ** 2)) ** 0.5
+            trip_cost = 2 * (distance * fuel_consumption / 100) * fuel_price
+            product_cost = shop.purchase(product_cart=self.product_cart)
+            total_cost = round((trip_cost + product_cost), 2)
+            self.trip_costs[total_cost] = shop
+            print(f"{self.name}'s trip to the {shop.name} costs {total_cost}")

```

Comment:
The code block inside the for loop could be refactored to avoid repetition and improve readability. Consider creating a separate method for calculating the trip cost and purchasing products from the shop.